scipy/stats/_qmc.py

```diff
@@ -1017,7 +1065,10 @@ def random(self, n: IntNumber = 1) -> np.ndarray:
             LHS sample.
 
         """
-        return self.lhs_method(n)
+        if self.optimization_method is None:
+            return self.lhs_method(n)
+        else:
+            return self.optimization_method(n)
```

Comment:
Minor point. Don't change it unless you really agree. I won't bring it up again:

I suggested it [here](https://github.com/scipy/scipy/pull/13471#discussion_r676155190), but this suggests even more to me now that the process is:
* generate a LHS
* If optimization is not None, optimize it.

Do you agree that is just a simplification of the logic:
* If optimization is None, 1) generate a LHS
* Otherwise, 1) generate a LHS and 2) optimize it

If someone had written it like this:
```python3
        if self.optimization_method is None:
            return self.lhs_method(n)
        else:
            lhs = self.lhs_method(n)
            return self.optimization_method(lhs)
```

You would think it should be simplified to:
```python3
        lhs = self.lhs_method(n)
        if self.optimization_method is not None:
            lhs = self.optimization_method(lhs)
        return lhs
```
no?

If so, then I don't think that the fact that `best_sample = self.lhs_method(n=n)` is hidden within the optimization_method should make a difference.

It seems conceptually simpler to me, and then there is no need for `best_sample = self.lhs_method(n=n)` in every optimization method. The function that does the optimization _just optimizes_, it doesn't doesn't generate the sample. The parts are modular; the roles are well-separated.

If you wanted to reduce line count, then above:
```python3
        optimization_method = {
            "random-cd": self._random_cd,
            None: lambda x: x
        }
```
And it becomes
```python3
        lhs = self.lhs_method(n)
        return self.optimization_method(lhs)
```
but I might label that as "clever" and less clear.

Reply:
I was in trouble with this one 😅 I tried hard to not pass anything to the random functions and I totally agree that doing so, I am hiding things in the optimization methods and this might not be good/clear.

Since you confirm this approach, I am happy to do this actually! Thanks.